src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -254,25 +257,51 @@ public RedirectView logout(final Map<String, Object> model) {
      * @should redirect to passwordReset view
      */
     @GetMapping(value = "/passwordReset")
-    public String getPasswordReset(@RequestParam("token") String token, @RequestParam("code") String code) {
-        return this.passwordReset(token, code);
+    public String getPasswordReset(@RequestParam("token") String token, @RequestParam("code") String code, Model model) {
+        return this.passwordReset(token, code, model);
     }
 
     /**
      * @should redirect to reset password page if token is valid
      * @should redirect to token expired page if token is invalid
+     * @should redirect to token expired page if token is expired
      */
     @PostMapping(value = "/passwordReset")
-    public String passwordReset(@RequestParam("token") String token, @RequestParam("code") String code) {
-        String nextPage = RESETPASSWORD_VIEW;
+    public String passwordReset(@RequestParam("token") String token, @RequestParam("code") String code, Model model) {
+        String nextPage = EXPIRED_PASSWORD_RESET_LINK_VIEW;
         try {
             spiService.validateResetPasswordToken(token, code);
-        } catch (Exception e) {
-            nextPage = EXPIRED_PASSWORD_RESET_LINK_VIEW;
+            nextPage = RESETPASSWORD_VIEW;
+        } catch (HttpClientErrorException e) {
+            if (e.getStatusCode().value() == 404)
+            {
+                model.addAttribute("forgotPasswordLink", buildUrlFromTheBody(e.getResponseBodyAsString()));
+            }
+        } catch (IOException e) {
+            //return expired password reset link view
         }
         return nextPage;
     }
 
+    private String buildUrlFromTheBody(String responseBodyAsString) {
+        final String baselink = "/reset/forgotpassword";
+        try {
+            uk.gov.hmcts.reform.idam.api.internal.model.ForgotPasswordRequest request = mapper.readValue(
```

Comment:
minor: not sure about using a model named ForgotPasswordRequest as a response for an error scenario

Reply:
It's an idea of returning the initial object of where the user was in the their journey

Reply:
same as https://github.com/hmcts/idam-api/pull/525#discussion_r350865675

we can have a chat with Nik tomorrow and see what we want this api service to look like